src/main/webapp/feedback/load-icon-feedback.js

```diff
@@ -0,0 +1,106 @@
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import TimestampUtil from '../timestamp-util.js';
+import IconFeedbackUtil from './icon-feedback-util.js';
+import ParsedIconFeedback from './parsed-icon-feedback.js';
+
+/* Handles loading IconFeedback from database and parses data for graph. */
+export default class LoadIconFeedback {
+  static #ENDPOINT_FEEDBACK = '/icon-feedback';
+  static #PARAM_LECTURE_ID = 'lectureId';
+
+  /* Each interval is 10 seconds, used to increment interval. */
+  static #INCREMENT_INTERVAL_MS = 10000;
+
+  #lectureId;
+  #parsedIconFeedback;
+
+  constructor(lectureId) {
+    this.#lectureId = lectureId;
+    this.#parsedIconFeedback = new ParsedIconFeedback();
+  }
+
+  async initialize() {
+    await this.loadIconFeedbackList();
+  }
+
+  /**
+   * Fetches available IconFeedback from `ENDPOINT_FEEDBACK`
+   * and parses the data for it to be graphed.
+   */
+  async loadIconFeedbackList() {
+    const url =
+        new URL(LoadIconFeedback.#ENDPOINT_FEEDBACK, window.location.origin);
+    url.searchParams.append(
+        LoadIconFeedback.#PARAM_LECTURE_ID, this.#lectureId);
+    const response = await fetch(url);
+    const jsonData = await response.json();
+    console.log(jsonData);
+    this.parseFeedback(jsonData);
+  }
+
+  /** Parses `iconFeedbackJson` data so that it can be graphed. */
+  parseFeedback(iconFeedbackJson) {
+    let intervalLowerBound = LoadIconFeedback.#INCREMENT_INTERVAL_MS;
+    let typeCountsAndInterval = this.makeCountDictionary(intervalLowerBound);
+    for (let index = 0; index < iconFeedbackJson.length;) {
+      const iconFeedback = iconFeedbackJson[index];
+      if (intervalLowerBound < iconFeedback.timestampMs &&
+          iconFeedback.timestampMs >
+              intervalLowerBound + LoadIconFeedback.#INCREMENT_INTERVAL_MS) {
+        this.#parsedIconFeedback.appendTypeCountsAndInterval(
+            typeCountsAndInterval);
+        intervalLowerBound += LoadIconFeedback.#INCREMENT_INTERVAL_MS;
+        typeCountsAndInterval = this.makeCountDictionary(intervalLowerBound);
+      } else {
+        const type = iconFeedback.type;
+        typeCountsAndInterval[type]++;
+        index++;
+      }
+    }
+    this.#parsedIconFeedback.appendTypeCountsAndInterval(typeCountsAndInterval);
+  }
+
+  /**
+   * Returns a dictionary mapping each icon type to a value which represents
+   * how many times that icon was clicked in an `intervalLowerBound`.
+   */
+  makeCountDictionary(intervalLowerBound) {
+    return {
+      [IconFeedbackUtil.TYPE_GOOD]: 0,
```

Comment:
I just realized how this dictionary is being used. This is not great. When you map X to Y then X is not 2 types. Icons and interval are different data types. I would create a class to wrap this information. The class could be called IntervalIconFeedbackCount (or some other name) and it has a time and a dictionary.

To keep this PR small, please make this change in a second PR which merged into master, then you can sync this PR and use that class to clean this up. Please let me know if you have any questions.